model/infinispan/src/main/java/org/keycloak/connections/infinispan/InfinispanMultiSiteLoadBalancerCheckProviderFactory.java

```diff
@@ -73,7 +75,7 @@ public String getId() {
 
     @Override
     public boolean isSupported(Config.Scope config) {
-        return Profile.isFeatureEnabled(Profile.Feature.MULTI_SITE) && !Profile.isFeatureEnabled(Profile.Feature.REMOTE_CACHE);
+        return MultiSiteUtils.isMultiSiteEnabled() && InfinispanUtils.isEmbeddedInfinispan();
```

Comment:
Shouldn't this always be enabled when MULTI_SITE feature is enabled as that now always implies REMOTE_CACHE?

Reply:
Good question. I thought this check did not make sense when the `REMOTE_CACHE` feature is enabled. However, when I look into the code, this checks also local caches, so maybe it makes sense to run this check too.

@pruivo, what was the reason for disabling this check when `REMOTE_CACHE` is used?  
```
Profile.isFeatureEnabled(Profile.Feature.MULTI_SITE) && !Profile.isFeatureEnabled(Profile.Feature.REMOTE_CACHE);
```

Reply:
I realized the `RemoteLoadBalancerCheckProviderFactory` is also checking embedded caches so I guess there is no need to enable `InfinispanMultiSiteLoadBalancerCheckProviderFactory.java`. Maybe we can remove that check?

Reply:
`InfinispanMultiSiteLoadBalancerCheckProviderFactory` checks the embedded caches only and `RemoteLoadBalancerCheckProviderFactory` checks the remote caches since the embedded caches do not exist.

Reply:
It checks also embedded local caches: https://github.com/keycloak/keycloak/blob/main/model/infinispan/src/main/java/org/keycloak/connections/infinispan/remote/RemoteLoadBalancerCheckProviderFactory.java#L138

Reply:
I created a PR for removing the old lb-check as I think it is no longer necessary:
https://github.com/keycloak/keycloak/pull/31834